src/python/bot/fuzzers/engine_common.py

```diff
@@ -62,6 +66,139 @@
 # Header format for logs.
 LOG_HEADER_FORMAT = ('Command: {command}\n' 'Bot: {bot}\n' 'Time ran: {time}\n')
 
+# Number of radamsa mutations.
+RADAMSA_MUTATIONS = 2000
+
+# Maximum number of seconds to run radamsa for.
+RADAMSA_TIMEOUT = 3
+
+
+class Generator(object):
+  """Generators we can use."""
+  NONE = 0
+  RADAMSA = 1
+  ML_RNN = 2
+
+
+def select_generator(strategy_pool, fuzzer_path):
+  """Pick a generator to generate new testcases before fuzzing or return
+  Generator.NONE if no generator selected."""
+  if environment.platform() == 'FUCHSIA':
+    # Unsupported.
+    return Generator.NONE
+
+  # We can't use radamsa binary on Windows. Disable ML for now until we know it
+  # works on Win.
+  # These generators don't produce testcases that LPM fuzzers can use.
+  if (environment.platform() == 'WINDOWS' or is_lpm_fuzz_target(fuzzer_path)):
+    return Generator.NONE
+  elif strategy_pool.do_strategy(strategy.CORPUS_MUTATION_ML_RNN_STRATEGY):
+    return Generator.ML_RNN
+  elif strategy_pool.do_strategy(strategy.CORPUS_MUTATION_RADAMSA_STRATEGY):
+    return Generator.RADAMSA
+
+  return Generator.NONE
+
+
+def generate_new_testcase_mutations(corpus_directory,
+                                    new_testcase_mutations_directory,
+                                    fuzzer_name, candidate_generator):
+  """Generate new testcase mutations, using existing corpus directory or other
+  methods."""
+  generation_timeout = get_new_testcase_mutations_timeout()
+
+  # Initialize to none, change if testcase mutations are succesfully generated
+  # according to radamsa or ml rnn.
+  generator_strategy = Generator.NONE
+  pre_mutations_filecount = shell.get_directory_file_count(
+      new_testcase_mutations_directory)
+
+  # Generate new testcase mutations using Radamsa.
+  if candidate_generator == Generator.RADAMSA:
+    generate_new_testcase_mutations_using_radamsa(
+        corpus_directory, new_testcase_mutations_directory, generation_timeout)
+  # Generate new testcase mutations using ML RNN model.
+  elif candidate_generator == Generator.ML_RNN:
+    generate_new_testcase_mutations_using_ml_rnn(
+        corpus_directory, new_testcase_mutations_directory, fuzzer_name,
+        generation_timeout)
+
+  # If new mutations are successfully generated, set generator strategy.
+  if shell.get_directory_file_count(
+      new_testcase_mutations_directory) > pre_mutations_filecount:
+    generator_strategy = candidate_generator
```

Comment:
It seems a little weird for this function to return a Generator value based on if it succeeded or not. How about just return True/False, since the caller knows what it passed?

Reply:
Sure